src/Npgsql/NpgsqlTypes/NpgsqlRange.cs

```diff
@@ -19,109 +20,237 @@
 // AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
 // ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
 // TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+
 #endregion
 
 using System;
+using System.ComponentModel;
+using System.Globalization;
 using System.Text;
 using JetBrains.Annotations;
-
-#pragma warning disable 1591
+using Microsoft.Extensions.Primitives;
 
 // ReSharper disable once CheckNamespace
 namespace NpgsqlTypes
 {
-    public struct NpgsqlRange<T> : IEquatable<NpgsqlRange<T>>
+    /// <summary>
+    /// Represents a PostgreSQL range type.
+    /// </summary>
+    /// <typeparam name="T">The element type of the values in the range.</typeparam>
+    /// <remarks>
+    /// See: https://www.postgresql.org/docs/current/static/rangetypes.html
+    /// </remarks>
+    [TypeConverter(typeof(NpgsqlRange<>.RangeTypeConverter))]
+    public readonly struct NpgsqlRange<T> : IEquatable<NpgsqlRange<T>>
     {
-#pragma warning disable CA1000
-        public static NpgsqlRange<T> Empty { get; } = new NpgsqlRange<T>(default, default, RangeFlags.Empty);
-#pragma warning restore CA1000
+        // -----------------------------------------------------------------------------------------------
+        // Regarding bitwise flag checks via @roji:
+        //
+        // > Note that Flags.HasFlag() used to be very inefficient compared to simply doing the
+        // > bit operation - this is why I've always avoided it. .NET Core 2.1 adds JIT intrinstics
+        // > for this, making Enum.HasFlag() fast, but I honestly don't see the value over just doing
+        // > a bitwise and operation, which would also be fast under .NET Core 2.0 and .NET Framework.
+        //
+        // See:
+        //   - https://github.com/npgsql/npgsql/pull/1939#pullrequestreview-121308396
+        //   - https://blogs.msdn.microsoft.com/dotnet/2018/04/18/performance-improvements-in-net-core-2-1
+        // -----------------------------------------------------------------------------------------------
+
+        /// <summary>
+        /// Defined by PostgreSQL to represent an empty range.
+        /// </summary>
+        const string EmptyLiteral = "empty";
```

Comment:
I personally don't see huge value in having these strings as constants, but I don't have much against it either :) OK to leave like this if you prefer.

Reply:
I think there could be some value in documenting that these literals are based on PostgreSQL, rather than Npgsql. But I can use inline comments for the same effect. Let me know if you change your mind on keep/remove.